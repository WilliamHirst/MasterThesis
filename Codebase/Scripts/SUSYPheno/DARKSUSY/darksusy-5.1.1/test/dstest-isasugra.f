      program dstest
c
c     This program tests some DarkSUSY routines
c
c     In addition to the DarkSUSY programs and data files, this test
c     program chooses a few mSUGRA models randomly and uses Isasugra to
c     calculate the low energy models. These are then fed into DarkSUSY
c     and the relic density is calculated.
c     If you want to calculate rates and stuff, use the calls as
c     outlined in dstest.f

c     
c-----This line is 72 columns long--------------------------------------
c
      implicit none
      real*8 oh2,xf,dsrdomega                            ! relic density
      character*80 message,scr
      logical first
      data first/.true./
      integer ii, jj, idum,nfc,iwar,ierr,unphys,hwarning
      real*8 m0min,m0max,m5min,m5max,a0min,a0max,tbmin,tbmax,
     &  m0,mhf,a0,tgb,sgnmu
      real*8 dsrndlin,dsrndsgn,dsabsq
c
c     Here we include the file dsmssm.h which contains common block
c     definitions for particle masses, susy parameters, and results.
c     We also include dsidtag.h which gives the possibility to tag
c     every model with a 12 character id tag (variable idtag). This
c     id tag is printed in case of errors for that model.
c
      include 'dsmssm.h'
      include 'dsio.h'
      include 'dsidtag.h'

      idum = -363469   ! seed for random number generator

c
c     This call initializes the DarkSUSY package. This call should
c     be the first call in any program using DarkSUSY.
c
      call dsinit
      call aldata   ! to make sure isajet common block datas are set
                    ! on most compilers, this is not needed

c
c     The amount of output onto standard output can be controlled by the
c     variable prtlevel. Setting prtlevel=0 suppresses all messages,
c     except fatal errors that terminate the program. Setting prtlevel=1
c     displays informational messages. Setting prtlevel to higher values
c     displays more and more information. In this example, we set
c     prtlevel=1 to have a message from dsreadpar. We reset it to 0
c     later. Had we set prtlevel=2 here, we would also have messages for
c     each variable which is (re)set in dsreadpar. Caution: if prtlevel
c     is reset in dsreadpar, the new value applies to subsequent
c     messages.
c
      prtlevel=1
c
c     This is an example of redefining DarkSUSY parameters reading them
c     from an input file, in this case the file dsdefault.dat which
c     contains the default values (this is done for test purposes only,
c     because the default values are already included in the code). The
c     user can copy the file dsdefault.dat to a new file, change the
c     values of the variables, and read in the new file.  Not all
c     variables need to be assigned in the new file, but only those
c     whose value differs from the default value. Note: dsdefault.dat
c     resets prtlevel to 0, ie no messages.
c     Note: the dsdefault.dat is not upto date, but a call like the
c     following can be used to reset parameters to your own choices.
c
c      open (unit=10,file='dsdefault.dat',status='old')
c      call dsreadpar(10)
c      close (10)


c     Now we are ready to scan the supersymmetric parameter space. 
c     We here read some models from dstest-isasugra.mod, which
c     are a few of the Benchmark points in hep-ph/0306219
c
      

      open(unit=10,file='dstest-isasugra.mod',status='old',
     &  form='formatted')
      read(10,'(a)') scr ! read header lines
      read(10,'(a)') scr ! read header lines
      read(10,'(a)') scr ! read header lines


c     these are minimal SUGRA
 20   read(10,*,end=100) idtag,m0,mhf,a0,sgnmu,tgb
c 1001 format(1x,A12,3(1x,E14.8),1x,e7.1,5(1x,e14.8))

      write(*,*) ' '
      write(*,*) 'Model: ',idtag
      write(*,*) 'Parameters: ',m0,mhf,a0,sgnmu,tgb

c      Now define the model, see dsgive_model_isasugra how we transfer
c      these varibles to DarkSUSY common blocks
      call dsgive_model_isasugra(m0,mhf,a0,sgnmu,tgb)
         

c      Now set up the model by a call to dssusy_isasugra. dssusy_isasugra
c      calls isasugra and transfers the results to DarkSUSY common blocks.
c      DarkSUSY is then set-up for this model. Check the flags unphys
c      and hwarning which are non-zero if the model is not good.
      call dssusy_isasugra(unphys,hwarning)
       
      write(*,*) '   hwarning = ',hwarning
      write(*,*) '     unphys = ',unphys
      if (hwarning.eq.0.and.unphys.eq.0) then  ! Model OK

c
c     We now have all the masses and couplings calculated. Let's print
c     some out to see what they are. dsabsq is just the square of the
c     absolute value of it's complex*16 argument.

        write(*,*) '  Neutralino mass = ',mass(kn(1))
        write(*,*) '  Gaugino fraction = ',
     &    dsabsq(neunmx(1,1))+dsabsq(neunmx(1,2))
        write(*,*) '  H1 mass =  ',mass(kh1),width(kh1)
        write(*,*) '  H2 mass =  ',mass(kh2),width(kh2)
        write(*,*) '  H3 mass =  ',mass(kh3),width(kh3)
        write(*,*) '  H+- mass = ',mass(khc),width(khc)

c...Now calculate the relic density with all coannihilations
        write(*,*) '   Calculating omega h^2, please be patient...'
        oh2=dsrdomega(1,1,xf,ierr,iwar,nfc)
        write(*,*) '   Oh2 = ',oh2

c...Go on calculating rates etc, as shown in dstest.f if we want to
      endif

      goto 20

 100  continue

      close(10)
      stop
      end
