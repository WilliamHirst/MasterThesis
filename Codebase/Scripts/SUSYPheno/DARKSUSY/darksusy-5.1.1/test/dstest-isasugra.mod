#Type:    3
# Benchmark models from Battaglia et al, hep-ph/0306219
#.....id..... ......m0...... ......mhf..... ......A0...... .sgnmu. ....tgbeta.... .unphys. ..valid.
 BA_AP_000001 0.10700000E+03 0.60000000E+03 0.00000000E+00 0.1E+01 0.50000000E+01        0        0
 BA_BP_000001 0.57000000E+02 0.25000000E+03 0.00000000E+00 0.1E+01 0.10000000E+02        0        0
 BA_CP_000001 0.80000000E+02 0.40000000E+03 0.00000000E+00 0.1E+01 0.10000000E+02        0        0
 BA_DP_000001 0.10100000E+03 0.52500000E+03 0.00000000E+00 -.1E+01 0.10000000E+02        0        0
 BA_EP_000001 0.15320000E+04 0.30000000E+03 0.00000000E+00 0.1E+01 0.10000000E+02        0        0
 BA_FP_000001 0.34400000E+04 0.10000000E+04 0.00000000E+00 0.1E+01 0.10000000E+02        0        0
 BA_GP_000001 0.11300000E+03 0.37500000E+03 0.00000000E+00 0.1E+01 0.20000000E+02        0        0
 BA_HP_000001 0.24400000E+03 0.93500000E+03 0.00000000E+00 0.1E+01 0.20000000E+02        0        0
 BA_IP_000001 0.18100000E+03 0.35000000E+03 0.00000000E+00 0.1E+01 0.35000000E+02        0        0
 BA_JP_000001 0.29900000E+03 0.75000000E+03 0.00000000E+00 0.1E+01 0.35000000E+02        0        0
 BA_KP_000001 0.10010000E+04 0.13000000E+04 0.00000000E+00 -.1E+01 0.46000000E+02        0        0
 BA_LP_000001 0.30300000E+03 0.45000000E+03 0.00000000E+00 0.1E+01 0.47000000E+02        0        0
 BA_MP_000001 0.11250000E+04 0.18400000E+04 0.00000000E+00 0.1E+01 0.51000000E+02        0        0
