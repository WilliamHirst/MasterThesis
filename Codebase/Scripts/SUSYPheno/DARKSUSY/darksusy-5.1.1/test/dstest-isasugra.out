 *********************************************************
 *** Welcome to DarkSUSY version                       ***
 *** trunk (rev 273)                                             ***
 *********************************************************
  
 Initializing DarkSUSY...
 WARNING in dspbset: using halo model nfwsm       
 for the setup. If you change halo model, make sure
 to call dspbset again.
 dsntsunread: Opening file 
 /data1/edsjo/DarkSUSY/library/trunk/share/DarkSUSY/bs05op.dat                  
                                                                                
                                           
 dsntsunread: Opening file 
 /data1/edsjo/DarkSUSY/library/trunk/share/DarkSUSY/nele_bs05op.dat             
                                                                                
                                           
 dsntsunread: tabulating potential inside the Sun...
   ...done
 dsntsunread: tabulating column density inside the Sun...
   ...done
 dsreadnuclides: reading /data1/edsjo/DarkSUSY/library/trunk/share/DarkSUSY/nuclides.dat
 Initialization of DarkSUSY complete.
  
 Model: BA_AP_000001
 Parameters:    107.000000000000        600.000000000000     
  0.000000000000000E+000   1.00000000000000        5.00000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    247.672531127930     
   Gaugino fraction =   0.994326464508866     
   H1 mass =     887.360107421875        2.12299880543452     
   H2 mass =     111.700912475586       2.922412818952870E-003
   H3 mass =     880.847717285156        2.34694363010867     
   H+- mass =    890.181945800781        2.50502919744649     
    Calculating omega h^2, please be patient...
 dsrdreaddof: reading /data1/edsjo/DarkSUSY/library/trunk/share/DarkSUSY/dsdofHP_B.dat
    Oh2 =   4.868351824846476E-002
  
 Model: BA_BP_000001
 Parameters:    57.0000000000000        250.000000000000     
  0.000000000000000E+000   1.00000000000000        10.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    96.7034912109375     
   Gaugino fraction =   0.971132073273594     
   H1 mass =     372.551513671875       0.859502252895855     
   H2 mass =     109.237915039062       3.431536645989221E-003
   H3 mass =     369.767974853516        1.13425089587723     
   H+- mass =    380.597564697266        1.14441340533216     
    Calculating omega h^2, please be patient...
    Oh2 =   0.119291601867765     
  
 Model: BA_CP_000001
 Parameters:    80.0000000000000        400.000000000000     
  0.000000000000000E+000   1.00000000000000        10.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    161.338165283203     
   Gaugino fraction =   0.988335426587633     
   H1 mass =     578.516418457031        1.30520146764102     
   H2 mass =     112.738777160645       3.110222601934334E-003
   H3 mass =     574.506835937500        1.46705774322936     
   H+- mass =    583.744995117188        1.87152928069632     
    Calculating omega h^2, please be patient...
    Oh2 =   8.646874312162373E-002
  
 Model: BA_DP_000001
 Parameters:    101.000000000000        525.000000000000     
  0.000000000000000E+000  -1.00000000000000        10.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    216.951751708984     
   Gaugino fraction =   0.994623111028156     
   H1 mass =     742.834411621094        1.66921370012090     
   H2 mass =     113.513626098633       3.078110569547553E-003
   H3 mass =     737.718627929688        1.74642058269952     
   H+- mass =    746.839843750000        2.40848329473406     
    Calculating omega h^2, please be patient...
    Oh2 =   5.719632401891166E-002
  
 Model: BA_EP_000001
 Parameters:    1532.00000000000        300.000000000000     
  0.000000000000000E+000   1.00000000000000        10.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    122.162635803223     
   Gaugino fraction =   0.967942070274590     
   H1 mass =     1566.51342773438        27.5995295742217     
   H2 mass =     112.973999023438       2.875855412094235E-003
   H3 mass =     1556.16015625000        28.0260115385509     
   H+- mass =    1568.44592285156        24.0523027620842     
    Calculating omega h^2, please be patient...
    Oh2 =    4.88640337956038     
  
 Model: BA_FP_000001
 Parameters:    3440.00000000000        1000.00000000000     
  0.000000000000000E+000   1.00000000000000        10.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    436.905029296875     
   Gaugino fraction =   0.992287116218655     
   H1 mass =     3571.73657226562        56.6759167246611     
   H2 mass =     118.131286621094       2.961769448217354E-003
   H3 mass =     3548.34130859375        58.1140540330061     
   H+- mass =    3572.58691406250        50.9121660348835     
    Calculating omega h^2, please be patient...
    Oh2 =    13.9647205693018     
  
 Model: BA_GP_000001
 Parameters:    113.000000000000        375.000000000000     
  0.000000000000000E+000   1.00000000000000        20.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    151.176452636719     
   Gaugino fraction =   0.987325954869500     
   H1 mass =     520.567993164062        3.82174135096586     
   H2 mass =     113.150650024414       3.129693287695937E-003
   H3 mass =     517.148010253906        3.91426154137027     
   H+- mass =    526.796203613281        5.41068479265515     
    Calculating omega h^2, please be patient...
    Oh2 =   0.109227887350407     
  
 Model: BA_HP_000001
 Parameters:    244.000000000000        935.000000000000     
  0.000000000000000E+000   1.00000000000000        20.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    396.256683349609     
   Gaugino fraction =   0.997400129922094     
   H1 mass =     1203.38720703125        7.93039862341401     
   H2 mass =     118.300994873047       3.018778466467606E-003
   H3 mass =     1195.50305175781        7.93117685103052     
   H+- mass =    1206.11340332031        12.4154063822455     
    Calculating omega h^2, please be patient...
    Oh2 =   0.129615052242506     
  
 Model: BA_IP_000001
 Parameters:    181.000000000000        350.000000000000     
  0.000000000000000E+000   1.00000000000000        35.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    141.097213745117     
   Gaugino fraction =   0.985663328427152     
   H1 mass =     443.154144287109        9.92808768824958     
   H2 mass =     112.770751953125       3.094269566644232E-003
   H3 mass =     440.289520263672        9.93094772900346     
   H+- mass =    451.001953125000        13.1568272286437     
    Calculating omega h^2, please be patient...
    Oh2 =   0.143602955491946     
  
 Model: BA_JP_000001
 Parameters:    299.000000000000        750.000000000000     
  0.000000000000000E+000   1.00000000000000        35.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    315.347045898438     
   Gaugino fraction =   0.996233847924564     
   H1 mass =     887.640808105469        18.1244251404865     
   H2 mass =     117.376396179199       3.016109218405656E-003
   H3 mass =     881.839294433594        18.0076745287891     
   H+- mass =    891.647705078125        27.8610736333055     
    Calculating omega h^2, please be patient...
    Oh2 =   0.122971526747246     
  
 Model: BA_KP_000001
 Parameters:    1001.00000000000        1300.00000000000     
  0.000000000000000E+000  -1.00000000000000        46.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    564.776733398438     
   Gaugino fraction =   0.998454318115590     
   H1 mass =     1028.44226074219        35.5369307376116     
   H2 mass =     119.570327758789       3.160047317157356E-003
   H3 mass =     1021.74652099609        35.3358889229084     
   H+- mass =    1032.58032226562        55.0502848708379     
    Calculating omega h^2, please be patient...
    Oh2 =    1.46131418569743     
  
 Model: BA_LP_000001
 Parameters:    303.000000000000        450.000000000000     
  0.000000000000000E+000   1.00000000000000        47.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    184.771499633789     
   Gaugino fraction =   0.990756207145561     
   H1 mass =     485.786987304688        19.3057909327959     
   H2 mass =     114.461097717285       3.021613071966994E-003
   H3 mass =     482.637939453125        19.1879089859308     
   H+- mass =    493.488922119141        26.3952316060926     
    Calculating omega h^2, please be patient...
    Oh2 =   0.146522951989762     
  
 Model: BA_MP_000001
 Parameters:    1125.00000000000        1840.00000000000     
  0.000000000000000E+000   1.00000000000000        51.0000000000000     
    hwarning =            0
      unphys =            0
   Neutralino mass =    809.178833007812     
   Gaugino fraction =   0.999101099918612     
   H1 mass =     1651.61889648438        66.3424524732756     
   H2 mass =     121.145576477051       3.046460263834189E-003
   H3 mass =     1640.82116699219        65.9603332100635     
   H+- mass =    1654.04174804688        104.515538609786     
    Calculating omega h^2, please be patient...
    Oh2 =   0.279867451256383     
